src/core/imageUtils.js

```diff
@@ -17,3 +17,11 @@ export const getPreviewImage = (addon, { index = 0, full = true } = {}) => {
   }
   return null;
 };
+
+export const getPreviewIndexBySize = (addon, size = 720) => {
+  const imageIndex =
+    addon.previews &&
+    addon.previews.findIndex((preview) => preview.image_size[0] === size);
```

Comment:
This line was confusing to me at first. I think it would help if you added a comment saying "preview.image_size[0] is the width"